cpp/src/arrow/compute/function_internal.cc

```diff
@@ -83,7 +83,8 @@ Result<std::unique_ptr<FunctionOptions>> GenericOptionsType::Deserialize(
 
 Result<std::unique_ptr<FunctionOptions>> DeserializeFunctionOptions(
     const Buffer& buffer) {
-  io::BufferReader stream(buffer);
+  // FIXME: Would ToString to heavy?
```

Comment:
I wonder would this be a bit heavy? Should I implement a `DeserializeFunctionOptions(const std::shared_ptr<Buffer>& buffer)`

Reply:
You can explicitly `std::make_shared<Buffer>(data_, nbytes)` to preserve the prior behavior.

Reply:
sorry I'm not fully understand. 

```c++
io::BufferReader stream(buffer.ToString());
```

is same as `io::BufferReader`. I mean should change `DeserializeFunctionOptions(const Buffer& buffer)`  to `DeserializeFunctionOptions(const std::shared_ptr<Buffer>& buffer)`

Reply:
What I mean is that you can keep `const Buffer&` and just explicitly wrap it in a non-owning `shared_ptr<Buffer>` to pass to `BufferReader`, which keeps the old behavior (zero-copy without owning)

Reply:
I got it, thanks!